### PR TITLE
Feature/collect subprojects

### PR DESCRIPTION
**Pull request by apmccartney**

### CODE REVIEW

File: collect_subprojects.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import shutil
+import sys
+import warnings
+
+def project_name():
+    return os.path.split( os.getcwd() )[1]
+
+def dependency_directory():
+    """
+    A function isolating the path specification to the project's dependency directory
```
Comment (jlconlin) [inline collect_subprojects.py:18]: I love your docstrings. Thanks!

File: collect_subprojects.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if os.path.isdir( dependency ) and not dependency in traversed :
+            traversed.add( dependency )
+            os.chdir( dependency )
+            if not os.path.isdir( os.path.join( destination, dependency ) ):
+                try:
+                    os.symlink( os.getcwd(),
+                                os.path.join( destination, dependency ) )
+
+                except OSError:
+                    warnings.warn( "Could not create symbolic link from {} to subprojects directory.".format( os.getcwd() ) )
```
Comment (jlconlin) [inline collect_subprojects.py]: This line is too long.

File: fetch_subprojects.py [hunk 3]
```diff
@@ -69,12 +69,15 @@ def traverse_dependencies( destination, traversed, git ):
             update_repository( git )
             if not os.path.isdir( os.path.join( destination, dependency ) ):
                 try:
-                    os.symlink( os.getcwd(), os.path.join( destination, dependency ) )
+                    os.symlink( os.getcwd(),
+                                os.path.join( destination, dependency ) )
 
                 except OSError:
                     warnings.warn( "Could not create symbolic link from {} to subprojects directory.".format( os.getcwd() ) )
```
Comment (jlconlin) [inline fetch_subprojects.py]: This line is also too long.

**Review by jlconlin — CHANGES_REQUESTED** [review]

**Comment by apmccartney** [comment]
Did you review the implementation?

File: collect_subprojects.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    return os.path.join( os.getcwd(), "dependencies" )
+
+def traverse_dependencies( destination, traversed ):
+    """
+    collect unique links to dependency projects in a destination folder
+    """
+    if not os.path.isdir( dependency_directory() ):
+        return
+    
+    os.chdir( dependency_directory() )
```
Comment (jlconlin) [inline collect_subprojects.py:29]: I always get nervous about using `chdir` because I can't keep track of what directory I'm in.

**Review by jlconlin — APPROVED** [review]

**Pull request merged**